wled00/mqtt.cpp

```diff
@@ -58,6 +65,9 @@ void onMqttMessage(char* topic, char* payload, AsyncMqttClientMessageProperties
   DEBUG_PRINT(F("MQTT msg: "));
   DEBUG_PRINTLN(topic);
 
+  // set connected to true. If we got a message, we must be connected (this fixes a lot of issues with AsyncMqttClient)
+  mqtt->setConnected(true); // note that setConnected is a function that I added to AsyncMqttClient
```

Comment:
This feels like a hack. Can you explain what issues you saw with the connected state?

Reply:
I often received messages but wasn't able to send a response because the client said it was disconnected, even though it clearly was connected since it got the messages successfully. This fixed that issue for me. It definitely is a bit hacky though.

Reply:
FYI I will veto this change (and corresponding AsyncMqttClient change) as it is inappropriate unless proved (with traces) MQTT client isn't working correctly.